pkg/heartbeat/sanitize.go

```diff
@@ -64,6 +68,15 @@ func Sanitize(h Heartbeat, config SanitizeConfig) Heartbeat {
 		h.Branch = nil
 	}
 
+	if h.EntityType == FileType && config.HideProjectFolder {
+		// normalize project path to look like a directory
+		if !strings.HasSuffix(h.ProjectPath, string(os.PathSeparator)) {
+			h.ProjectPath += string(os.PathSeparator)
+		}
+
+		h.Entity = strings.TrimPrefix(h.Entity, h.ProjectPath)
```

Comment:
We should actually trim both `--project-folder` and the auto detected project folder from `h.Entity` because someone might open a file from another project. The `--project-folder` wouldn't match, but the auto-detected folder would.